### PR TITLE
Daemon improvements

### PR DESCRIPTION
**Pull request by ngortheone**
Code readability and consistency improvements. The change is broken down into small commits, each commit has a commit message with rationale for the change.

cc @kevans91 @bsdimp

All changes pass kyua tests defined for daemon

P.S. this is a re-incarnation of https://github.com/freebsd/freebsd-src/pull/667, I messed up that branch and Github refused to re-open that PR.

### CODE REVIEW

**Review by igalic — APPROVED** [review]
I think you can mark this as ready for review

File: usr.sbin/daemon/daemon.c [hunk 1]
```diff
@@ -297,8 +298,8 @@ main(int argc, char *argv[])
 	supervision_enabled = pidfile  != NULL ||
 		ppidfile != NULL ||
 		restart  != 0    ||
-		outfd    != -1   ||
-		syslog_enabled == true;
+		logpar.outfd    != -1   ||
+		logpar. syslog_enabled == true;
```
Comment (igalic) [inline usr.sbin/daemon/daemon.c]: why is this `syslog_enabled` not in `logpar`?

**Review by igalic — COMMENTED** [review]
❓

File: usr.sbin/daemon/daemon.c (hunk 1 above)
Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: small typo, I fixed it in one of the following commits

Comment (ngortheone) [reply usr.sbin/daemon/daemon.c]: fixed

**Comment by ngortheone** [comment]
@kevans91 landed this one https://reviews.freebsd.org/rG6b4ef4b16a6930d91d0d99c4ceb75cd43a7526c1

**Pull request closed without merging**